src/Commands/GenerateModule.php

```diff
@@ -0,0 +1,321 @@
+        $this->menuHook($io);
+
+        $hookStringInstall="";
+        foreach (self::$tabHook as $hookname) {
+            $hookStringInstall .= "
+            && \$this->registerHook('".$hookname."')";
+        }
+        $hookStringFunction="";
+        foreach (self::$tabHook as $hookfunction) {
+            $hookfunction = ucfirst($hookfunction);
+            $hookStringFunction .= "
+    public function hook".$hookfunction."(\$params) {
+        //TODO
+    }
+    ";
+        }
+
+        $stringEntryPoint = "";
+        $stringEntryPoint.= "<?php
+if (!defined('_CAN_LOAD_FILES_')) {
+    exit;
+}
+
+require_once __DIR__ . '/vendor/autoload.php';
+
+use PrestaShop\PrestaShop\Core\Module\WidgetInterface;
+
+/**
+* Class ".$filenameFirstletterCaps."
+*/
+
+class ".$filenameFirstletterCaps." extends Module implements WidgetInterface
+{
+
+    const MODULE_NAME = '".$fileName."';
+
+    public \$templateFile;
+
+    public function __construct()
+    {
+        \$this->name = '".$filenameCamel."';
+        \$this->author = '".$author."';
+        \$this->version = '1.0.0';
+        \$this->need_instance = 0;
+        \$this->bootstrap = true;
+        parent::__construct();
+
+        \$this->displayName = '".$filenameCamel."';
+        \$this->description = '".$description."';
+        \$this->secure_key = Tools::encrypt(\$this->name);
+
+        \$this->ps_versions_compliancy = array('min' => '1.7.5.0', 'max' => _PS_VERSION_);
+        \$this->templateFile = 'module:".$filenameCamel."/views/templates/hook/';
+
+        \$this->tabs = [
+        ];
+    }
+
+    public function install()
+    {
+        return parent::install()".$hookStringInstall."
+            && \$this->installTab();
+    }
+
+    public function uninstall()
+    {
+        return parent::uninstall()
+            && \$this->uninstallTab();
+    }
+
+    public function installTab()
+    {
+        //TODO
+    }
+
+    public function uninstallTab()
+    {
+        //TODO
+    }
+
+    public function enable(\$force_all = false)
+    {
+        return parent::enable(\$force_all)
+            && \$this->installTab();
+    }
+
+    public function disable(\$force_all = false)
+    {
+        return parent::disable(\$force_all)
+            && \$this->uninstallTab();
+    }
+    ".$hookStringFunction."
+    public function renderWidget(\$hookName, array \$configuration)
+    {
+        //TODO
+    }
+
+    public function getWidgetVariables(\$hookName, array \$configuration)
+    {
+        //TODO
+    }
+}";
+        $this->whriteCode($filesystem, $fileName, $stringEntryPoint);
+
+//        $io->text("table 1");
+//        $io->table(
+//            ['Nom', 'Type', 'Null'],
+//            [
+//                ['id_truc', 'int(11)', 'Y'],
+//                ['date_bidule', 'datetime', 'Y'],
+//                ['machin', 'varchar(20)', 'Y'],
+//            ]
+//        );
+    }
+
+    protected function whriteCode($filesystem, $fileName, $string)
```

Comment:
i think you mean `write`.
Also your method is more writing in a file than writing code, the fact that it is code shouldn't be related to the method name IMO.
This method don't do much different as dumpFile does so you should maybe add some descriptive words on what it does differently like `dumpModulePHPFile` ?

Reply:
Function renamed and commented as well as the menuHook Function